lib/presentation/widget/shared_space_document/shared_space_document_viewmodel.dart

```diff
@@ -0,0 +1,846 @@
+import 'package:linshare_flutter_app/presentation/view/header/simple_bottom_sheet_header_builder.dart';
+import 'package:linshare_flutter_app/presentation/view/modal_sheets/confirm_modal_sheet_builder.dart';
+import 'package:linshare_flutter_app/presentation/view/modal_sheets/edit_text_modal_sheet_builder.dart';
+import 'package:linshare_flutter_app/presentation/view/order_by/order_by_dialog_bottom_sheet.dart';
+import 'package:linshare_flutter_app/presentation/widget/base/base_viewmodel.dart';
+import 'package:linshare_flutter_app/presentation/widget/destination_picker/destination_picker_action/copy_destination_picker_action.dart';
+import 'package:linshare_flutter_app/presentation/widget/destination_picker/destination_picker_action/negative_destination_picker_action.dart';
+import 'package:linshare_flutter_app/presentation/widget/destination_picker/destination_picker_arguments.dart';
+import 'package:linshare_flutter_app/presentation/widget/shared_space_document/shared_space_document_arguments.dart';
+import 'package:linshare_flutter_app/presentation/widget/shared_space_document/shared_space_document_type.dart';
+import 'package:linshare_flutter_app/presentation/widget/shared_space_document/shared_space_document_ui_type.dart';
+import 'package:linshare_flutter_app/presentation/widget/upload_file/destination_type.dart';
+import 'package:linshare_flutter_app/presentation/widget/upload_file/upload_file_arguments.dart';
+import 'package:open_file/open_file.dart' as open_file;
+import 'package:permission_handler/permission_handler.dart';
+import 'package:redux/src/store.dart';
+import 'package:redux_thunk/redux_thunk.dart';
+import 'package:share/share.dart' as share_library;
+
+class SharedSpaceDocumentNodeViewModel extends BaseViewModel {
+  final AppNavigation _appNavigation;
+  final LocalFilePicker _localFilePicker;
+  final VerifyNameInteractor _verifyNameInteractor;
+  final GetAllChildNodesInteractor _getAllChildNodesInteractor;
+  final CreateSharedSpaceFolderInteractor _createSharedSpaceFolderInteractor;
+  final GetSorterInteractor _getSorterInteractor;
+  final SaveSorterInteractor _saveSorterInteractor;
+  final SortInteractor _sortInteractor;
+  final RenameSharedSpaceNodeInteractor _renameSharedSpaceNodeInteractor;
+  final SearchWorkGroupNodeInteractor _searchWorkGroupNodeInteractor;
+  final DownloadWorkGroupNodeInteractor _downloadWorkGroupNodeInteractor;
+  final DownloadPreviewWorkGroupDocumentInteractor _downloadPreviewWorkGroupDocumentInteractor;
+  final DownloadMultipleNodeIOSInteractor _downloadMultipleNodeIOSInteractor;
+  final CopyMultipleFilesToMySpaceInteractor _copyMultipleToMySpaceInteractor;
+  final CopyMultipleFilesToSharedSpaceInteractor _copyMultipleFilesToSharedSpaceInteractor;
+  final RemoveMultipleSharedSpaceNodesInteractor _removeMultipleSharedSpaceNodesInteractor;
+
+  StreamSubscription _storeStreamSubscription;
```

Comment:
why we need that?

Reply:
It is used to listen to workGoupNode searches on SharedSpace